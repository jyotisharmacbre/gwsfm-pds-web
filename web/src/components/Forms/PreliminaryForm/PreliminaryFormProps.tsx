import EventType from "../../../enums/EventType";

export interface Props {
	onSave: (saveAll: boolean, event: EventType, prelimComponentDetails: any, index: number) => void;
	onPrevious: () => void;
	onToggle: (id: string) => void;
	preliminariesDetails: any;
	currencySymbol: string;
	isExpand: boolean;
	componentIdList: Array<string>;
	countryCode: string;
	insuranceRate: number;
	preliminaryState: any;
	subContractorState: any;
	discountState: any;
	projectStatus: number;
	event: EventType;
	loading: boolean;
}