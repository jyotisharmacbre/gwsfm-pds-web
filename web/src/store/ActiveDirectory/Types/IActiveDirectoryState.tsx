import { IAdHOPData, IAdPMData, IAdPOData } from './IActiveDirectory';

export interface IActiveDirectoryState {
  ADhopData: Array<IAdHOPData>;
  ADpmData: Array<IAdPMData>;
  ADpoData: Array<IAdPOData>;
  error: string | null;
}
