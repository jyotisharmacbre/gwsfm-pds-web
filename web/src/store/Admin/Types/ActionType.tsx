export enum ActionType {
	GET_PROJECT_PARAMETERS_SUCCESS = 'GET_PROJECT_PARAMETERS_SUCCESS',
	GET_PROJECT_PARAMETERS_ERROR = 'GET_PROJECT_PARAMETERS_ERROR'
}
