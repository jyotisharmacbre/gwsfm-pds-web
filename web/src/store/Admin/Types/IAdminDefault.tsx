export interface IAdminDefaults {
    projectParameterId: number;
    countryId: number;
    name: string;
    value: string;
  }
  