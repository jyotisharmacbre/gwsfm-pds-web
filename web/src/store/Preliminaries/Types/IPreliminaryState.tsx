import { IPreliminariesComponentDetails } from './IPreliminariesComponentDetails';
import Notify from '../../../enums/Notify';

export interface IPreliminaryForm {
  preliminaryDetails: Array<IPreliminariesComponentDetails>;
 
}

export interface IPreliminaryState {
  projectId: string;
  preliminaryDetails:Array<IPreliminariesComponentDetails>;
}
