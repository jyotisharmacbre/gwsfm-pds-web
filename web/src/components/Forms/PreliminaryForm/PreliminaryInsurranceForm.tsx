import React from 'react';
import { Field, formValueSelector } from 'redux-form';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import { Validate, alphaNumeric, onlyNumber } from '../../../helpers/fieldValidations';
import { restrictMinus, restrictMinusAndDecimal} from '../../../helpers/utility-helper';
import {calculateSell, calculateTotalCost} from '../../../helpers/formulas';
import { IState } from '../../../store/state';
import { connect } from 'react-redux';
import { ICurrency } from '../../../store/Lookups/Types/ICurrency';
import { IPreliminariesComponentDetails } from '../../../store/Preliminaries/Types/IPreliminariesComponentDetails';
import { getPreliminarySummaryCalculation } from '../../../helpers/pricing-calculation-helper';
interface Props {
  fields:any;
      itemDetail:any;
      currencies:Array<ICurrency>|null;
      currencyId:number;
      currencySymbol:string;
      componentIndex:number;
      preliminaryData:Array<IPreliminariesComponentDetails>;
}  

const PreliminaryInsurranceForm:React.FC<Props>
= (props: Props) =>{
 
  const{fields,itemDetail,componentIndex,currencies,currencyId,currencySymbol,preliminaryData}=props;
  const updateCost=(index:number)=>
  {
    props.preliminaryData[props.componentIndex].items[index].totalCost=calculateTotalCost(getPreliminarySummaryCalculation(props.preliminaryData).cost);
  }
  return (
  <tbody>
  {props.fields.map((member, index) => (
   <tr key={index}>
     <td>
       <Field
                        name={`${member}.nameOfSupplier`}
                        data-test="nameOfSupplier"
                        type="text"
                        className="width-120 mb-0"
                        component={PdsFormInput}
                        validate={[
                            Validate.maxLength(1000),
                            alphaNumeric
                          ]}
                      />
    </td>
  <td>
    <Field
                name={`${member}.totalCost`}
                type="number"
                normalize={restrictMinus}
                input={{
                  value:calculateTotalCost(getPreliminarySummaryCalculation(props.preliminaryData).cost),
                    disabled: true,
                    onchange: updateCost(index)
                  
                  }}
                component={PdsFormInput}
                className="width-120 pl-20 required currency"
                validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  
                ]}
                currency={currencySymbol}
                divPosition="relative"
              />
    </td>
    <td>
    <Field
                  name={`${member}.grossMargin`}
                  type="number"
                  normalize={restrictMinusAndDecimal}
                  component={PdsFormInput}
                  className="width-120 pl-20 required currency"
                  validate={[
                      Validate.maxLength(15),
                      onlyNumber
                    
                  ]}
                  currency={"%"}
                  divPosition="relative"
                />
    </td>
    <td>
    <Field
                  name={'totalSell'}
                  type="text"
                  input={{
                     value:calculateSell(props.preliminaryData[props.componentIndex].items[index].totalCost,props.preliminaryData[props.componentIndex].items[index].grossMargin),
                     disabled: true 
                    }}
                   component={PdsFormInput}
                  className="width-120 pl-20 required currency"
                  validate={[
                      Validate.maxLength(15),
                      onlyNumber
                  ]}
                  currency={currencySymbol}
                  divPosition="relative"
                />
    </td>
    <td>
    <Field
                        name={`${member}.comments`}
                        data-test="comments"
                        type="text"
                        component={PdsFormInput}
                        validate={[
                            Validate.maxLength(1000),
                            alphaNumeric
                          ]}
                      />
    </td>

     </tr>
    ))}
    </tbody>
  )};
  const mapStateToProps = (state: IState) => ({
    preliminaryData: selector(state,"preliminaryDetails")
   
  });
 const selector=formValueSelector("PreliminaryForm");
export default connect(mapStateToProps)(PreliminaryInsurranceForm);;

