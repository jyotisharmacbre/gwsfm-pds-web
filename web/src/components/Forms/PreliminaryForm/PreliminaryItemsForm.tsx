import React from 'react';
import { Field, formValueSelector } from 'redux-form';
import PdsFormInput from '../../PdsFormHandlers/PdsFormInput';
import { Validate, alphaNumeric, onlyNumber } from '../../../helpers/fieldValidations';
import { getFilterElementFromArray,calculateCost,calculateSell} from '../../../helpers/utility-helper';
import { IState } from '../../../store/state';
import { connect } from 'react-redux';
import { ICurrency } from '../../../store/Lookups/Types/ICurrency';
import { IPreliminariesComponentDetails } from '../../../store/Preliminaries/Types/IPreliminariesComponentDetails';
interface Props {
  fields:any;
      itemDetail:any;
      currencies:Array<ICurrency>|null;
      currencyId:number;
      currencySymbol:string;
      componentIndex:number;
      preliminaryData:Array<IPreliminariesComponentDetails>;
}  

const PreliminaryItemsForm:React.FC<Props>
= (props: Props) =>{
  const updateCost=(index:number)=>
  {
    props.preliminaryData[props.componentIndex].items[index].totalCost=(props.preliminaryData[props.componentIndex].items[index].noOfHours*props.preliminaryData[props.componentIndex].items[index].hourRate);
  }
  const{fields,itemDetail,componentIndex,currencies,currencyId,currencySymbol,preliminaryData}=props;
  const currencySymbols:string=getFilterElementFromArray(props.currencies,"currencyId",props.currencyId,"currencySymbol");

  return (
  <tbody>
  {props.fields.map((member, index) => (
   <tr>
     <td>{props.itemDetail.items[index].itemName}</td>
     <td>
                       {props.itemDetail.items[index].itemId=="2"||props.itemDetail.items[index].itemId=="3"?<Field
                        name={`${member}.nameOfSupplier`}
                        input={{disabled: true} }
                        data-test="nameOfSupplier"
                        type="text"
                        component={PdsFormInput}
                        validate={[
                            Validate.maxLength(1000),
                            alphaNumeric
                          ]}
                      />:<Field
                      name={`${member}.nameOfSupplier`}
                      data-test="nameOfSupplier"
                      type="text"
                      component={PdsFormInput}
                      validate={[
                          Validate.maxLength(1000),
                          alphaNumeric
                        ]}
                    />}
    </td>
    <td>
    {(props.itemDetail.items[index].itemId=="1"||props.itemDetail.items[index].itemId=="2")?<Field
                  name={`${member}.noOfHours`}
                  type="number"
                 input={{ disabled: true}}
                  component={PdsFormInput}
                  className="width-120 pl-20 required "
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  divPosition="relative"
                />:<Field
                name={`${member}.noOfHours`}
                type="number"
                component={PdsFormInput}
                className="width-120 pl-20 required "
                validate={[
                  Validate.maxLength(15),
                  onlyNumber
                ]}
                divPosition="relative"
              />}
    </td>
     <td>
                {(props.itemDetail.items[index].itemId=="1"||props.itemDetail.items[index].itemId=="2")?<Field
                  name={`${member}.hourRate`}
                  type="number"
                 input={{ disabled: true}}
                  component={PdsFormInput}
                  className="width-120 pl-20 required "
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  currency={currencySymbols}
                  divPosition="relative"
                  placeholderKey=""
                />:<Field
                name={`${member}.hourRate`}
                type="number"
                component={PdsFormInput}
                className="width-120 pl-20 required "
                validate={[
                  Validate.maxLength(15),
                  onlyNumber
                ]}
                currency={currencySymbols}
                divPosition="relative"
                placeholderKey=""
              />}
    </td>
  <td>
    {props.itemDetail.items[index].itemId!="3"&&props.itemDetail.items[index].itemId!="4"?<Field
                  name={`${member}.totalCost`}
                  type="number"
                  component={PdsFormInput}
                  className="width-120 pl-20 required"
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  currency={currencySymbols}
                  divPosition="relative"
                />:<Field
                name={`${member}.totalCost`}
                type="number"
                input={{
                  value:calculateCost(props.preliminaryData[props.componentIndex].items[index].noOfHours,props.preliminaryData[props.componentIndex].items[index].hourRate),
                  disabled: true,
                  onchange:updateCost(index)
                  }}
                component={PdsFormInput}
                className="width-120 pl-20 required"
                validate={[
                  Validate.maxLength(15),
                  onlyNumber
                ]}
                currency={currencySymbols}
                divPosition="relative"
              />}
    </td>
    <td>
    <Field
                  name={`${member}.grossMargin`}
                  type="number"
                  component={PdsFormInput}
                  className="width-120 pl-20 required"
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  currency={"%"}
                  divPosition="relative"
                />
    </td>
    <td>
    <Field
                  name={'totalSell'}
                  type="text"
                  input={{
                    value:calculateSell(props.preliminaryData[props.componentIndex].items[index].totalCost,props.preliminaryData[props.componentIndex].items[index].grossMargin),
                     disabled: true 
                    }}
                   component={PdsFormInput}
                  className="width-120 pl-20 required"
                  validate={[
                    Validate.maxLength(15),
                    onlyNumber
                  ]}
                  currency={currencySymbols}
                  divPosition="relative"
                />
    </td>
    <td>
    <Field
                        name={`${member}.comments`}
                        data-test="comments"
                        type="text"
                        component={PdsFormInput}
                        validate={[
                            Validate.maxLength(1000),
                            alphaNumeric
                          ]}
                      />
    </td>

     </tr>
    ))}
    </tbody>
  )};
  const mapStateToProps = (state: IState) => ({
    preliminaryData: selector(state,"preliminaryDetails"),
    currencies:state.lookup.currencies,
    currencyId:state.project.form.currencyId
  });
 const selector=formValueSelector("PreliminaryForm");
export default connect(mapStateToProps)(PreliminaryItemsForm);;

