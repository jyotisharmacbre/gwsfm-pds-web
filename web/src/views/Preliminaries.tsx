import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import { IState } from '../store/state';
import { IPreliminariesComponentDetails } from '../store/Preliminaries/Types/IPreliminariesComponentDetails';
import * as actions from '../store/rootActions';
import PreliminaryForm from '../components/Forms/PreliminaryForm/PreliminaryForm';
import { IPreliminaries } from '../store/Preliminaries/Types/IPreliminaries';
import { convertIntoDatabaseModel } from "../store/Preliminaries/DataWrapper";
import { toast } from 'react-toastify';
import Notify from '../enums/Notify';
import { ILookup } from '../store/Lookups/Types/ILookup';
import { ICurrency } from '../store/Lookups/Types/ICurrency';
import { FormattedMessage } from 'react-intl';
import EventType from '../enums/EventType';
import CalculationsSummaryTable from '../components/Table/CalculationsSummaryTable';
import CalculationsSummaryType from '../enums/CalculationsSummaryType';
import { getPropertyName,getFilterElementFromArray } from '../helpers/utility-helper';
import { formValueSelector } from 'redux-form';
import Currency from '../store/Lookups/InitialState/Currency';
import { History } from 'history';
interface IMapStateToProps {
  preliminaryDetails: Array<IPreliminariesComponentDetails>;
  lookupData: Array<ILookup>;
  currencies: Array<ICurrency> | null;
  match: any;
  notify: Notify;
  event: EventType;
  currencyId: number;
  preliminaryForm:Array<IPreliminariesComponentDetails>;
  history:History;
}
interface IMapDispatchToProps {
  preliminaryAdd: (
    preliminaryDetails: Array<IPreliminaries>,
    event: EventType
  ) => void;
  preliminaryEdit: (
    preliminaryDetails: Array<IPreliminaries>,
    event: EventType
  ) => void;
  getPreliminaryDetails: (projectId: string) => void;
  updateInputField: (inputData: any) => void;
  getAllCurrencies: () => void;
  getProjectStatus: () => void;
  getProjectDetail: (projectId: string) => void;
}

const Preliminaries: React.FC<
  IMapStateToProps & IMapDispatchToProps
> = props => {
  let paramProjectId: string = '';
  const CurrencyObj = new Currency();
  let isLookupSessionExists: boolean = (sessionStorage.getItem("lookupData") != null && sessionStorage.getItem("lookupData") != undefined && sessionStorage.getItem("lookupData") != "")
  useEffect(() => {
    window.scrollTo(0, 0);
    paramProjectId=props.match.params.projectId;
    props.getProjectDetail(paramProjectId);
    props.getProjectStatus();
    props.getAllCurrencies();
    if (paramProjectId != null && paramProjectId != '' && paramProjectId != undefined && isLookupSessionExists) {
      props.getPreliminaryDetails(paramProjectId);
    }
  }, []);
  useEffect(() => {
    window.scrollTo(0, 0);
    if (props.notify == Notify.success) {
      toast.success('Data Saved Successfully');
      if (props.event == EventType.next) {
        props.history.push('/Subcontractor/' + props.match.params.projectId);
      }
    }
    else if (props.notify == Notify.error) {
      toast.error('Error occured.Please contact to administrator.');
    }

  }, [props.notify, props.event]);
  useEffect(() => {
    if (paramProjectId != null && paramProjectId != '' && paramProjectId != undefined && isLookupSessionExists) {
      props.getPreliminaryDetails(paramProjectId);
    }
  }, [props.lookupData]);
  const handleExpandAllEvent = () => {
    var element: any = document.getElementsByClassName('expandAll');
    for (let i = 0; i < element.length; i++) {
      element[i].classList.remove('hide');
      element[i].classList.add('show');
    }
  };
  const handleToggle = (id: string) => {
    var element: any = document.getElementById('collapse_' + id);
    if (element != null) {
      var isClassExists = element.classList.contains('show');
      if (isClassExists) {
        element.classList.add('hide');
        element.classList.remove('show');
      } else {
        element.classList.remove('hide');
        element.classList.add('show');
      }
    }
  };
  const handleSaveData = (
    saveAll: boolean,
    event: EventType,
    preliminaryDetails: any,
    index: number

  ) => {
    var editData: Array<IPreliminaries> = [];
    var saveData: Array<IPreliminaries> = [];
    paramProjectId = props.match.params.projectId;
    let preData: Array<IPreliminariesComponentDetails> = [];
    preData.push(preliminaryDetails.preliminaryDetails[index]);
    var preliminariesData: Array<IPreliminaries> = saveAll ?
      convertIntoDatabaseModel(preliminaryDetails.preliminaryDetails, paramProjectId) :
      convertIntoDatabaseModel(preData, paramProjectId)
    editData = preliminariesData.filter((data) => {
      return (data.TotalCost > 0 && data.PreliminaryId != '');
    })
    saveData = preliminariesData.filter((data) => {
      return (data.TotalCost > 0 && data.PreliminaryId == '');
    })
    if (editData.length > 0 && saveData.length > 0) { props.preliminaryEdit(editData, event); props.preliminaryAdd(saveData, event); }
    else if (editData.length > 0) { props.preliminaryEdit(editData, event); }
    else if (saveData.length > 0) { props.preliminaryAdd(saveData, event) }
    else { toast.error('No data changed to save.'); }
  };
  const handlePrevious = () => {
    props.history.push(`/Project/${props.match.params.projectId}`);
  };

  return (
    <div className="container-fluid">
      <div className=" row">
        <div className="col-lg-12">
          <form className="custom-wrap">
            <div className="heading-subtitle">
              <h1>
                <span className="d-md-block d-none">
                  <FormattedMessage id='TITLE_JUSTIFICATION'></FormattedMessage>
                </span>
                <span className="d-md-none"> <FormattedMessage id='TITLE_JUSTIFICATION_SHORT'></FormattedMessage></span>
              </h1>
              <p className="text-green"> <FormattedMessage id='PAGE_PRELIMINARY_TITLE'></FormattedMessage></p>
            </div>

            <div className="row">
              <div className="col-lg-12">
                <div className="row">
                  <div className="col-lg-8">
                    <div className="table-responsive">
                      <CalculationsSummaryTable
                      projectId={props.match.params.projectId}
                      name={CalculationsSummaryType.preliminary} 
                      preliminary={props.preliminaryForm}
                      currencySymbol={getFilterElementFromArray(
                        props.currencies,
                        getPropertyName(
                        CurrencyObj,
                        prop => prop.currencyId
                      ),
                        props.currencyId,
                        getPropertyName(
                        CurrencyObj,
                        prop => prop.currencySymbol
                      )
                      )}/>
                    </div>
                  </div>
                  <div className="col-lg-4">
                    <div className="text-right">
                      <button type="button" className="active fltRght" onClick={() => handleExpandAllEvent()}>EXPAND ALL</button>
                    </div>
                  </div>
                </div>
              </div>
            </div>

            <div>
              <PreliminaryForm
                onSave={handleSaveData}
                onPrevious={handlePrevious}
                onToggle={handleToggle}
                preliminariesDetails={props.preliminaryDetails}
                currencySymbol={getFilterElementFromArray(
                    props.currencies,
                    getPropertyName(
                    CurrencyObj,
                    prop => prop.currencyId
                  ),
                    props.currencyId,
                    getPropertyName(
                    CurrencyObj,
                    prop => prop.currencySymbol
                  )
                  )}
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

const selector = formValueSelector('PreliminaryForm');


const mapStateToProps = (state: IState) => {
  return {
    preliminaryDetails: state.preliminary.preliminaryDetails,
    lookupData: state.lookup.projectstatus,
    currencies: state.lookup.currencies,
    notify: state.preliminary.notify,
    currencyId: state.project.form.currencyId,
    event: state.preliminary.event,
    preliminaryForm: selector(state, 'preliminaryDetails')
  };
};

const mapDispatchToProps = dispatch => {
  return {
    preliminaryAdd: (preliminaryDetails, event) =>
      dispatch(actions.preliminaryAdd(preliminaryDetails, event)),
    preliminaryEdit: (preliminaryDetails, event) =>
      dispatch(actions.preliminaryEdit(preliminaryDetails, event)),
    getProjectDetail: (projectId: string) =>
      dispatch(actions.getProjectDetail(projectId)),
    getPreliminaryDetails: (projectId: string) =>
      dispatch(actions.getPreliminaryDetails(projectId)),
    getAllCurrencies: () => dispatch(actions.getAllCurrencies()),
    getProjectStatus: () => dispatch(actions.getProjectStatus())
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Preliminaries);
