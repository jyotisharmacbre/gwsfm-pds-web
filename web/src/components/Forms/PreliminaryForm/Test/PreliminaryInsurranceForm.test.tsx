import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import translations from '../../../../Translations/translation';
import configureStore from 'redux-mock-store';
import { reduxForm } from 'redux-form';
import { preliminariuserData, currencies } from './PreliminaryFormTestData';
import PreliminaryItemsForm from '../PreliminaryItemsForm';
import { IPreliminaryForm } from '../../../../store/Preliminaries/Types/IPreliminaryState';
import { Props } from '../PreliminaryFormProps';
import { lookUpInitialState } from '../../ProjectOverviewForm/Test/ActivityFeedListTestData';
import { preliminariesData } from '../../../../views/Test/Preliminaries/PreliminariesTestData';
import PreliminaryInsurranceForm from '../PreliminaryInsurranceForm';

let wrapper;
const mockStore = configureStore([]);
let store;
const Decorated = reduxForm<IPreliminaryForm, Props>({ form: "PreliminaryForm" })(PreliminaryInsurranceForm);
const mockDBStore = () => {
	store = mockStore({
		form: {
			PreliminaryForm: {
				values: {
					preliminaryDetails: preliminariuserData
				}
			}
		},
		preliminary: preliminariuserData,
		
	});
};

const mountComponent = (fields, itemDetail, currencies, currencyId, currencySymbol, componentIndex, preliminaryData) => {
	wrapper = mount(
		<Provider store={store}>
			<IntlProvider locale="en" messages={translations['en'].messages}>
				<Decorated
					itemDetail={itemDetail}
					currencies={currencies}
					currencyId={currencyId}
					currencySymbol={currencySymbol}
					componentIndex={componentIndex}
					fields={[...fields]}
				/>
			</IntlProvider>
		</Provider>
	);
};

describe('Preliminary Items Form Form testCases', () => {
	beforeEach(() => {
		mockDBStore();
		let first = JSON.parse(JSON.stringify(preliminariuserData));
		let second = JSON.parse(JSON.stringify(preliminariuserData));
		mountComponent(first[0].items, second[0], currencies, 1, '$', 0, preliminariuserData);
	});

	it('Defines the component', () => {
		expect(wrapper).toBeDefined();
	});
	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
