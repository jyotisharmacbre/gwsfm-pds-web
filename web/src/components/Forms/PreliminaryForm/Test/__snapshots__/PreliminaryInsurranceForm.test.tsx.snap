// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preliminary Items Form Form testCases should match the snapshot 1`] = `ReactWrapper {}`;
